conformance/lang/expressions/new-expr/stream_class_descriptor.balt

```diff
@@ -0,0 +1,317 @@
+}
+
+Test-Case: error
+Description: Test new-expression to create a stream with incompatible stream type parameters.
+Labels: explicit-new-expr, stream-type-parameters, string
+
+function errorFunction() {
+    stream<int> _ = new stream<string>(); // @error incompatible types: expected 'stream<int>', found 'stream<string>'
+}
+
+Test-Case: output
+Description: Test new-expression to create a stream with incompatible type-descriptor.
+Labels: error-type, explicit-new-expr, optional-type, stream, stream-type-parameters, string
+
+function init() {
+    stream s = new stream<string, error?>();
+    io:println(s); // @output stream <stream<string,error?>>
+}
+
+Test-Case: output
+Description: Test new-expression to create a empty streams.
+Labels: error-type, explicit-new-expr, is-expr, method-call-expr, module-type-defn, nil-type, optional-type,
+        record-type, stream-type-parameters, union-type
+
+type IntRecord record {|
+    int value;
+|};
+
+function init() {
+    stream<int, error?> emptyStream = new stream<int, error?>();
+    IntRecord|error? d = emptyStream.next();
+    io:println(d is ()); // @output true
+}
+
+Test-Case: output
+Description: Test new-expr to create a stream without completion type.
+Labels: explicit-new-expr, is-expr, method-call-expr, nil-type, optional-type, record-type, stream-type-parameters
+
+type IntRecord record {|
+    int value;
+|};
+
+function init() {
+    stream<int> emptyStream = new stream<int>();
+    IntRecord? d = emptyStream.next();
+    io:println(d is ()); // @output true
+}
+
+Test-Case: error
+Description: Test new-expr to create a stream with stream extra parameters.
+Labels: error-type, explicit-new-expr, implicit-new-expr, isolated-method, member-access-expr, module-class-defn,
+        optional-type, stream-type-parameters, string
+
+class NumbersGenerator {
+    int i = 0;
+    public isolated function next() returns string|error? {
+        self.i += 1;
+        return self.i.toBalString();
+    }
+}
+
+function errorFunction() {
+    NumbersGenerator numbersGenerator = new;
+    stream<string, error?> numberStream = new stream<string, error?>(numbersGenerator, string); // @error 'new(numbersGenerator, null)' is not a valid constructor for streams type
+}
+
+Test-Case: error
+Description: Test new-expr to create a stream with stream implementor with next method with incompatible return type.
+Labels: error-type, explicit-new-expr, implicit-new-expr, isolated-method, member-access-expr, module-class-defn,
+        optional-type, stream-type-parameters
+
+class NumbersGenerator {
+    int i = 0;
+    public isolated function next() returns string|error? {
+        self.i += 1;
+        return self.i.toBalString();
+    }
+}
+
+function errorFunction() {
+    NumbersGenerator numbersGenerator = new;
+    stream<int, error?> numberStream = new stream<int, error?>(numbersGenerator); // @error incompatible types: expected '(IntRecord|ErrorB)?', found '(record {| int value; |}|ErrorA)?'
```

Comment:
Here again the error message is not clear, right? We should use the proper error to assert and also create issues for jBallerina to improve the errors.